src/main/kotlin/app/revanced/patches/shared/settings/patch/AbstractSettingsResourcePatch.kt

```diff
@@ -0,0 +1,137 @@
+package app.revanced.patches.shared.settings.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.DomFileEditor
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.*
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patches.shared.settings.BasePreference
+import app.revanced.patches.shared.settings.IResource
+import app.revanced.patches.shared.settings.addPreference
+import app.revanced.patches.shared.settings.addResource
+import app.revanced.patches.shared.settings.impl.ArrayResource
+import app.revanced.patches.shared.settings.impl.StringResource
+import app.revanced.patches.youtube.misc.manifest.patch.FixLocaleConfigErrorPatch
+import app.revanced.util.resources.ResourceUtils
+import app.revanced.util.resources.ResourceUtils.copyResources
+import org.w3c.dom.Node
+
+/**
+ * Abstract settings resource patch
+ *
+ * @param revancedPreferencesName Name of the settings preference xml file
+ * @param revancedPreferencesSourceDir Source directory to copy the preference template from
+ */
+@Description("Applies mandatory patches to implement ReVanced settings into the application.")
+@Version("0.0.1")
+@DependsOn([FixLocaleConfigErrorPatch::class])
+abstract class AbstractSettingsResourcePatch(
+    val revancedPreferencesName: String,
+    val revancedPreferencesSourceDir: String,
+) : ResourcePatch {
+    override fun execute(context: ResourceContext): PatchResult {
+        /* used for self-restart */
+        context.xmlEditor["AndroidManifest.xml"].use { editor ->
+            editor.file.getElementsByTagName("manifest").item(0).also {
+                it.appendChild(it.ownerDocument.createElement("uses-permission").also { element ->
+                    element.setAttribute("android:name", "android.permission.SCHEDULE_EXACT_ALARM")
```

Comment:
I am not sure, but I think this is only required for YouTube.

Reply:
I also need it for the Twitch patch; however, it isn't necessarily required by future settings patches for other apps, so I moved it back to YouTube's class.